plugins/azure/fix_plugin_azure/resource/containerservice.py

```diff
@@ -919,11 +917,31 @@ class AzureManagedCluster(AzureResource):
     workload_auto_scaler_profile: Optional[AzureManagedClusterWorkloadAutoScalerProfile] = field(default=None, metadata={'description': 'Workload Auto-scaler profile for the managed cluster.'})  # fmt: skip
 
     def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
-        from fix_plugin_azure.resource.compute import AzureDiskEncryptionSet
+        from fix_plugin_azure.resource.compute import AzureDiskEncryptionSet, AzureVirtualMachineScaleSet
 
         if disk_id := self.disk_encryption_set_id:
             builder.add_edge(self, edge_type=EdgeType.default, clazz=AzureDiskEncryptionSet, id=disk_id)
 
+        if agent_pool_profiles := self.agent_pool_profiles:
+            vmss_agent_pool_names_and_ids = self._get_poolnames_and_vmss_ids(builder)
+            for agent_pool_profile_name in agent_pool_profiles:
+                for info in vmss_agent_pool_names_and_ids:
+                    pool_name, vmss_id = info
+                    if agent_pool_profile_name == pool_name:
+                        builder.add_edge(
+                            self, edge_type=EdgeType.default, clazz=AzureVirtualMachineScaleSet, id=vmss_id
+                        )
+
+    def _get_poolnames_and_vmss_ids(self, builder: GraphBuilder) -> List[Tuple[str, str]]:
+        from fix_plugin_azure.resource.compute import AzureVirtualMachineScaleSet
+
+        get_poolname: Callable[[AzureVirtualMachineScaleSet], str] = (
+            lambda scale_set: scale_set.tags.get("aks-managed-poolName") or ""
+        )
+        get_vmss_id: Callable[[AzureVirtualMachineScaleSet], str] = lambda scale_set: scale_set.id or ""
+
+        return [(get_poolname(vmss), get_vmss_id(vmss)) for vmss in builder.nodes(clazz=AzureVirtualMachineScaleSet)]
```

Comment:
```suggestion
        return [(poolname, vmssid) for vmss in builder.nodes(clazz=AzureVirtualMachineScaleSet) if (poolname:=....) and (vmss_id:=...)]
```
Let's try avoiding `or ""` - I can see potential bugs.
You can simply assign and check in python.